src/main/kotlin/nl/avisi/structurizr/site/generatr/site/C4PlantUmlExporterWithElementLinks.kt

```diff
@@ -0,0 +1,51 @@
+package nl.avisi.structurizr.site.generatr.site
+
+import com.structurizr.Workspace
+import com.structurizr.export.IndentingWriter
+import com.structurizr.export.plantuml.C4PlantUMLExporter
+import com.structurizr.model.Element
+import com.structurizr.model.SoftwareSystem
+import com.structurizr.view.View
+import nl.avisi.structurizr.site.generatr.includedSoftwareSystems
+import nl.avisi.structurizr.site.generatr.normalize
+
+class C4PlantUmlExporterWithElementLinks(
+    private val workspace: Workspace,
+    private val branch: String
+): C4PlantUMLExporter() {
+    companion object {
+        const val TEMP_URI = "https://will-be-changed-to-relative"
+    }
+
+    override fun writeElement(view: View?, element: Element?, writer: IndentingWriter?) {
+        if (element !is SoftwareSystem || element.linkNotNeeded(view))
+            return super.writeElement(view, element, writer)
+
+        setElementUrl(element)
+        writeModifiedElement(view, element, writer)
+        restoreElement(element)
+    }
+
+    private fun Element.linkNotNeeded(view: View?) =
```

Comment:
I would invert the boolean logic, and rename this to `linkNeeded`. Booleans which have a negation in their name are often confusing to work with.